### PR TITLE
Re-approve when another review is requested

### LINKED ISSUE #206: Action does not perform review when requesting for re-review
**Steps to reproduce:**
1. Setup action, create a PR, and have it auto reviewed
2. Request for **re-review** in PR, **without** further code changes, review status should be something like this
image
3. Trigger `auto-approve-action`

**Actual output:**
Does not do a re-review, but logs a warning: `Current user already approved pull request #31, nothing to do`, despite the PR needing a re-review

**Expected output:** 
`auto-approve-action` should perform a re-review as the status is "Awaiting requested review...", if re-review is successful, status should be updated to "Approved"

Here's a sample PR: https://github.com/vincejv/fpi-framework/pull/31
image

### PR DESCRIPTION
**Pull request by hmarr**
When a review has already been left for the current commit, but the review is dismissed and re-requested, we should leave another approving review.

This also switches uses `Promise.all` to fetch the PR, review, and review requests concurrently.

Closes #206.

### CODE REVIEW

File: src/approve.ts [hunk 1]
```diff
@@ -30,43 +30,43 @@ export async function approve(
...
     core.info(`Getting pull request #${prNumber} info`);
-    const pull_request = await client.rest.pulls.get({
-      owner: context.repo.owner,
-      repo: context.repo.repo,
-      pull_number: prNumber,
-    });
-    const commit = pull_request.data.head.sha;
+    const { owner, repo } = context.repo;
+    const queryParams = { owner, repo, pull_number: prNumber };
+    const [pullRequest, reviews, requestedReviewers] = await Promise.all([
```
Comment (vincejv) [inline src/approve.ts:35]: Hi @hmarr, while doing the fixes, I realized we can get `requested_reviewers` from `pull_request`, I was about to PR this, so no additional rest calls should be necessary, if my understanding is correct with how Github API works

```javascript
for (const review of reviews.data) {
      if (
        review.user?.login == login &&
        review.commit_id == commit &&
        review.state == "APPROVED" &&
        pull_request.data
        .requested_reviewers?.filter(
          reviewer => reviewer.login == login
        ).length == 0
      ) {
        core.info(
          `Current user already approved pull request #${prNumber}, nothing to do`
        );
        return;
      }
    }
```

Comment (hmarr) [reply src/approve.ts:35]: Oh great point! Please do change that and submit the PR. I'd definitely rather avoid the extra request.

Comment (vincejv) [reply src/approve.ts:35]: > Oh great point! Please do change that and submit the PR. I'd definitely rather avoid the extra request.

The code is vastly different now, so I recommend for you to do the change 😁 on how you like it, but just suggesting that we can get the field without the additional rest call.

Comment (vincejv) [reply src/approve.ts:35]: @hmarr nevermind I PRd https://github.com/hmarr/auto-approve-action/pull/210, let me know your thoughts...

Comment (hmarr) [reply src/approve.ts:35]: Great, I'll close this out in favour of your pull request. Thanks for contributing!

**Pull request closed without merging**